pkg/operator/staticpod/certsyncpod/certsync_cmd.go

```diff
@@ -50,36 +55,65 @@ func NewCertSyncControllerCommand(configmaps, secrets []revision.RevisionResourc
 	return cmd
 }
 
-func (o *CertSyncControllerOptions) Complete() (*CertSyncController, error) {
-	kubeConfig, err := client.GetKubeConfigOrInClusterConfig(o.KubeConfigFile, nil)
+func (o *CertSyncControllerOptions) Run() error {
+	// When the kubeconfig content change, commit suicide to reload its content.
+	observer, err := fileobserver.NewObserver(500 * time.Second)
 	if err != nil {
-		return nil, err
+		return err
 	}
-	protoKubeConfig := rest.CopyConfig(kubeConfig)
-	protoKubeConfig.AcceptContentTypes = "application/vnd.kubernetes.protobuf,application/json"
-	protoKubeConfig.ContentType = "application/vnd.kubernetes.protobuf"
 
-	// This kube client use protobuf, do not use it for CR
-	kubeClient, err := kubernetes.NewForConfig(protoKubeConfig)
-	if err != nil {
-		return nil, err
-	}
-	kubeInformers := informers.NewSharedInformerFactoryWithOptions(kubeClient, 10*time.Minute, informers.WithNamespace(o.Namespace))
+	initialContent, _ := ioutil.ReadFile(o.KubeConfigFile)
+	observer.AddReactor(fileobserver.ExitOnChangeReactor, map[string][]byte{o.KubeConfigFile: initialContent}, o.KubeConfigFile)
+
+	stopCh := make(chan struct{})
```

Comment:
why not move it into a param of `Run`, while touching this code.

Reply:
normally, CLI commands does not pass chans via Run(), just following conventions.